syntax = "proto3";

package model;

message Request {
    int64 client_id = 1;
    string request_data = 2;
    repeated float float_values = 3;
}

message Response {
    int64 server_id = 1;
    string response_data = 2;
    repeated string results_model = 3;
}

service GRPCModelService {
    rpc SimpleRPC (Request) returns (Response);
    rpc RequestStreamingMethod (stream Request) returns (Response);
    rpc ResponseStreamingMethod (Request) returns (stream Response);
    rpc BidirectionalStreamingMethod (stream Request) returns (stream Response);
}